### ISSUE TITLE
Dealing with NNG_ENOTSUP in socket.c

### ISSUE THREAD

**Issue by ProfFan** [nanomsg/nng#610]
Note that by filing an issue, you agree that you have read and agreed to
our contribution guidelines.

## NNG & Platform details.

master ccc24a8, macOS 10.13.4

## Expected Behavior

`pubsub` and `pair1` tests should not fail.

## Actual Behavior

`pubsub` and `pair1` tests failed.

## Steps to Reproduce

```
cmake .. -DNNG_ENABLE_HTTP=OFF -DNNG_ENABLE_TLS=OFF -DNNG_PROTO_BUS0=OFF -DNNG_PROTO_PAIR0=ON -DNNG_PROTO_PULL0=OFF -DNNG_PROTO_PUSH0=OFF -DNNG_PROTO_REP0=OFF  -DNNG_PROTO_REQ0=OFF -DNNG_PROTO_RESPONDENT0=OFF -DNNG_PROTO_SURVEYOR0=OFF -DNNG_TRANSPORT_IPC=OFF -DNNG_TRANSPORT_TCP=OFF -DNNG_TRANSPORT_WS=OFF
make -j
make test
```

## Dirty Solution

Change `socket.c` at line 495:

```
	    ((rv = nni_sock_setopt(s, NNG_OPT_RECVMAXSZ, &s->s_rcvmaxsz,
	          sizeof(size_t), NNI_TYPE_SIZE)) != 0)) {
		sock_destroy(s);
		return (rv);
	}
```
To:
```
	    ((rv = nni_sock_setopt(s, NNG_OPT_RECVMAXSZ, &s->s_rcvmaxsz,
	          sizeof(size_t), NNI_TYPE_SIZE)) != 0)) {
            if(rv != NNG_ENOTSUP){
                sock_destroy(s);
                return (rv);
            } else {
                rv = 0;
            }
	}
```

### Before

```
54% tests passed, 17 tests failed out of 37

Total Test time (real) =   4.46 sec

The following tests FAILED:
	  1 - aio (Failed)
	  2 - bufsz (Failed)
	  3 - device (Failed)
	 10 - multistress (Child aborted)
	 15 - pollfd (Failed)
	 18 - scalability (SEGFAULT)
	 19 - sock (Failed)
	 21 - transport (Failed)
	 24 - pair1 (Failed)
	 25 - pubsub (Failed)
	 26 - cplusplus_pair (Child aborted)
	 27 - inproc_lat (Failed)
	 28 - inproc_thr (Failed)
	 29 - nngcat_async (Failed)
	 35 - nngcat_pubsub (Failed)
	 36 - nngcat_recvmaxsz (Failed)
	 37 - nngcat_unlimited (Failed)
Errors while running CTest
make: *** [test] Error 8
```

### After

```
78% tests passed, 8 tests failed out of 37

Total Test time (real) =  11.50 sec

The following tests FAILED:
	 10 - multistress (SEGFAULT)
	 18 - scalability (SEGFAULT)
	 19 - sock (Failed)
	 21 - transport (Failed)
	 29 - nngcat_async (Failed)
	 35 - nngcat_pubsub (Failed)
	 36 - nngcat_recvmaxsz (Failed)
	 37 - nngcat_unlimited (Failed)
Errors while running CTest
make: *** [test] Error 8
```

**Comment by gdamore** [comment]
Thanks,  I'll have to fix those tests to simply not run or not use those protocols.

**Comment by ProfFan** [comment]
@gdamore I think it is necessary to check for ENOTSUP as a special case, as ENOTSUP will not prevent a socket connection from happening.

**Comment by gdamore** [comment]
Looking at this in more detail, it looks like the problem is that you have disabled all the transports except inproc.  Unfrotunately, inproc does not yet honor this option, and that's an outstanding bug.

**Comment by gdamore** [comment]
See bug #541

**Comment by gdamore** [comment]
I have confirmed this, and since fixed it -- and verified that the fix for #541 addresses this problem properly.
